operator/src/main/java/org/bf2/operator/managers/StrimziBundleManager.java

```diff
@@ -200,10 +200,23 @@ private Approval approveInstallation(Subscription subscription) {
             return Approval.UNKNOWN;
         }
 
-        // CRDs are not installed, nothing we can do more ... just approving installation
-        if (!this.isKafkaCrdsInstalled()) {
+        final String subNamespace = subscription.getMetadata().getNamespace();
+        final String subName = subscription.getMetadata().getName();
+        boolean approveImmediately;
+
+        if (subscription.getStatus().getInstalledCSV() == null) {
+            log.infof("Subscription %s/%s has no linked CSV; InstallPlan will be approved immediately", subNamespace, subName);
+            approveImmediately = true;
+        } else if (!this.isKafkaCrdsInstalled()) {
+            log.infof("Subscription %s/%s has missing Strimzi CRDs; InstallPlan will be approved immediately", subNamespace, subName);
+            approveImmediately = true;
+        } else {
+            approveImmediately = false;
```

Comment:
Minor nit, just initialize approveImmediately as false.

Reply:
Done, latest commit also removes some modifiers in the unit test class Sonar flags as smells.